src/useMap.tsx

```diff
@@ -0,0 +1,87 @@
+import React from 'react'
+
+import { MapkitContext } from './MapkitProvider'
+import { MapOptions, propsToMapConstructionOptions } from './utils'
+
+import {
+  NumberTuple,
+  Rect,
+  RegionType,
+  createCoordinate,
+  createMapRect,
+  createCoordinateRegionFromValues,
+} from './utils'
+
+export const useMap = (defaultOptions: MapOptions = {}) => {
+  const [defaultMapOptions] = React.useState(defaultOptions)
+  let { mapkit } = React.useContext(MapkitContext)
+  let mapRef = React.useRef<HTMLDivElement>(null)
+  let [map, setMap] = React.useState<mapkit.Map>()
+
+  React.useEffect(() => {
+    if (mapkit && mapRef.current) {
```

Comment:
Why do you need to do this if check? Shouldn't the ref be valid after mount?

Reply:
Typescript :)